baseapp/grpcserver.go

```diff
@@ -33,7 +34,14 @@ func (app *BaseApp) RegisterGRPCServer(server gogogrpc.Server) {
 		if heightHeaders := md.Get(grpctypes.GRPCBlockHeightHeader); len(heightHeaders) > 0 {
 			height, err = strconv.ParseInt(heightHeaders[0], 10, 64)
 			if err != nil {
-				return nil, err
+				return nil, sdkerrors.Wrapf(
+					sdkerrors.ErrInvalidRequest,
+					"Baseapp.RegisterGRPCServer: invalid height header %q: %v", grpctypes.GRPCBlockHeightHeader, err)
+			}
+			if height < 0 {
+				return nil, sdkerrors.Wrapf(
+					sdkerrors.ErrInvalidRequest,
+					"Baseapp.RegisterGRPCServer: invalid height header %q: value out of range, must be >= 0", grpctypes.GRPCBlockHeightHeader)
```

Comment:
This function calls `createQueryContext` just below. Do you think it's worth having the negative height check here as well?

Reply:
In (*BaseApp).createQueryContext we perform that check https://github.com/cosmos/cosmos-sdk/pull/7663/files#diff-389d02f089ab55575400f5300d3c5cfe277fde5f04bd0fe6a85456a7252d1099R543 and that's part of this change.

Reply:
Exactly, we're performing the check twice. I think we can remove one.

Reply:
Yes, no need to duplicate. Let's remove the code above, or even use the `uint64` type (better)

Reply:
++. Let's remove the dup check

Reply:
I think the check from L59 is also duplicate